src/set/set.service.ts

```diff
@@ -1,345 +1,418 @@
-import { InjectModel } from '@nestjs/mongoose';
-import { Model, ObjectId, Types } from 'mongoose';
-import { JwtUserDto } from '../auth/dto/jwt.dto';
-import { Status } from '../shared/enums/status.enum';
-import { SharedService } from '../shared/shared.service';
-import { Role } from '../user/enums/role.enum';
-import { CreateSetDto } from './dto/create-set.dto';
-import { CreateTaskDto } from './dto/create-task.dto';
-import { UpdateSetDto } from './dto/update-set.dto';
-import { UpdateTaskDto } from './dto/update-task.dto';
-import { SetDocument } from './entities/set.entity';
-import { TaskDocument } from './entities/task.entity';
-import { TaskType } from './enums/tasktype.enum';
-import { SetSampleData } from './set.data';
-import { ResponseSet, ResponseSetMetadata, ResponseSetWithTasks, ResponseTask, ResponseTaskWithStatus, UpdatedCounts } from './types/set.response';
+import {
+    ForbiddenException,
+    Injectable,
+    InternalServerErrorException,
+    NotFoundException,
+} from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Model, ObjectId, Types } from 'mongoose'
+import { JwtUserDto } from '../auth/dto/jwt.dto'
+import { Status } from '../shared/enums/status.enum'
+import { Role } from '../user/enums/role.enum'
+import { CreateSetDto } from './dto/create-set.dto'
+import { CreateTaskDto } from './dto/create-task.dto'
+import { UpdateSetDto } from './dto/update-set.dto'
+import { UpdateTaskDto } from './dto/update-task.dto'
+import { SetDocument } from './entities/set.entity'
+import { TaskDocument } from './entities/task.entity'
+import { TaskType } from './enums/tasktype.enum'
+import { SetSampleData } from './set.data'
+import {
+    ResponseSet,
+    ResponseSetMetadata,
+    ResponseSetWithTasks,
+    ResponseTask,
+    ResponseTaskWithStatus,
+    UpdatedCounts,
+} from './types/set.response'
 
 @Injectable()
 export class SetService {
-  constructor(
-    @InjectModel('Set') private setSchema: Model<SetDocument>,
-    @InjectModel('Task') private taskSchema: Model<TaskDocument>,
-    private readonly sharedService: SharedService
-  ) { }
-
-
-  async createSet(createSetDto: CreateSetDto, user: JwtUserDto): Promise<ResponseSet> {
-    try {
-
-      const set: SetDocument = await this.setSchema.create({ ...createSetDto, createdBy: user.userId })
-
-      return {
-        _id: set.id,
-        daresCount: set.daresCount,
-        truthCount: set.truthCount,
-        language: set.language,
-        name: set.name,
-        createdBy: {
-          _id: user.userId,
-          username: user.username
-        },
-        previewImage: set.previewImage,
-        bannerImage: set.bannerImage
-      }
-
-    } catch (error) {
-      if (error.code == '11000') {
-        throw new ConflictException('This set already exists')
-      }
-      console.error(error)
-      throw new InternalServerErrorException()
+    constructor(
+        @InjectModel('Set') private setSchema: Model<SetDocument>,
```

Comment:
Same thing with the strings...you might want to replace this

Reply:
Yes